_posts/2020-04-01-#18401.md

```diff
@@ -0,0 +1,133 @@
+---
+layout: pr
+date: 2020-04-01
+title: "Initialize PrecomputedTransactionData in CheckInputScripts and Move single-sig checking EvalScript code to EvalChecksig"
+pr: 18401
+authors: [sipa]
+components: [consensus]
+host: jnewbery
+status: upcoming
+commit:
+---
+
+This week, we'll review two (small) PRs:
+
+- [PR 18401: Initialize PrecomputedTransactionData in
+  CheckInputScripts](https://github.com/bitcoin/bitcoin/pull/18401)
+- [PR 18422: Move single-sig checking EvalScript code to
+  EvalChecksig](https://github.com/bitcoin/bitcoin/pull/18422).
+
+## Notes
+
+- A few weeks ago, [we looked at PRs 16902 and
+  18002](https://bitcoincore.reviews/16902.html), which had both been pulled
+  out of the [WIP schnorr/taproot
+  implementation](https://github.com/bitcoin/bitcoin/pull/17977). This week,
+  we'll look at two more small refactor PRs that have also been pulled out of
+  that PR.
+
+- The same general considerations apply to these PRs: ordinarily, these
+  refactor changes to consensus code wouldn't meet the bar for review cost and
+  risk -vs- benefit.  However, splitting them off from the main schnorr/taproot
+  PR makes reviewing the logical changes to consensus in PR 17977 easier and
+  safer.
+
+### Initialize PrecomputedTransactionData in CheckInputScripts
```

Comment:
suggest title these two PR headers the same as at the top by prefixing with "PR 18401: "